src/components/common/custom-bottom-bar/Tab.tsx

```diff
@@ -0,0 +1,97 @@
+import { memo, useMemo } from 'react';
+import { Pressable } from 'react-native';
+import Animated, {
+  interpolate,
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated';
+
+import { TabList } from '~app/(tabs)/_layout';
+import { Icon, Stack, Text } from '~components/uikit';
+import { styled } from '~styles';
+
+const ANIMATION_DURATION = 350;
+
+type TabBarButtonProps = {
+  onPress: () => void;
+  onLongPress: () => void;
+  isFocused: boolean;
+  label: string;
+  tab: TabList[number];
+};
+
+const TabBarButton = ({
+  isFocused,
+  label,
+  tab,
+  ...pressableProps
+}: TabBarButtonProps) => {
+  const iconScale = useSharedValue(isFocused ? 1 : 0);
+  const labelOpacity = useSharedValue(isFocused ? 1 : 0);
+
+  const updateFocusAnimation = (focused: boolean) => {
+    iconScale.value = withSpring(focused ? 1 : 0, {
+      duration: ANIMATION_DURATION,
+    });
+    labelOpacity.value = withSpring(focused ? 1 : 0, {
+      duration: ANIMATION_DURATION,
+    });
+  };
+
+  useMemo(() => updateFocusAnimation(isFocused), [isFocused]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Where the hell did that come from 🤯 

I had to do a major rebasing because this custom bar code was actually quite old and looks like I didn't re-read properly before pushing... I appreciate your Sauron's eye preventing this craziness from being merged